companion.py

```diff
@@ -128,6 +125,49 @@ def check_modules_ships(self) -> None:
             self.data['lastStarport']['ships'] = {'shipyard_list': {}, 'unavailable_list': []}
 
 
+class CAPIDataEncoder(json.JSONEncoder):
+    """Allow for json dumping via specified encoder."""
+
+    def default(self, o):
+        """Tell JSON encoder that we're actually just a dict."""
+        return o.__dict__
+
+
+class CAPIDataRawEndpoint:
+    """Last received CAPI response for a specific endpoint."""
+
+    def __init__(self, raw_data: str, query_time: datetime.datetime):
+        self.query_time = query_time
+        self.raw_data = raw_data
+        # TODO: Maybe requests.response status ?
+
+
+class CAPIDataRaw:
+    """The last obtained raw CAPI response for each endpoint."""
+
+    raw_data: Dict[str, CAPIDataRawEndpoint] = {}
+
+    def record_endpoint(
+            self, endpoint: str,
+            raw_data: str,
+            query_time: datetime.datetime = datetime.datetime.utcnow()
```

Comment:
Only evaluated at class parse time.

Reply:
Changed to mandatory, and set at the call site.